src/irmin/object_graph.ml

```diff
@@ -182,6 +191,41 @@ struct
     in
     visit ()
 
+  let iter_rev ~pred ~min ~max ~node ~skip () =
+    Log.debug (fun f -> f "iter on closure (%d elements)" (List.length max));
```

Comment:
I wondering if this should not be the default implementation for iter (and then users can use `skip` to store an hashtable). Need to run benchmarks to understand the exact impact. 

Note that this start to look very similar to the API proposed by @liautaud for the GC.

Reply:
The default iter now does not use the hash table and added an explicit [iter with tracking](https://github.com/mirage/irmin/pull/1122/files#diff-e838a79261906c2af6541f96fc7772f019e640f6e12820a8da7195cfc78e0256R321).